lib/vagrant/util/caps.rb

```diff
@@ -0,0 +1,58 @@
+require "tempfile"
+require "fileutils"
+require "pathname"
+require "vagrant/util/directory"
+require "vagrant/util/subprocess"
+
+module Vagrant
+  module Util
+    module Caps
+      module BuildISO
+
+        # Check that the host has the ability to generate ISOs
+        #
+        # @param [Vagrant::Environment] env
+        # @return [Boolean]
+        def isofs_available(env)
+          if !self.const_defined?(:BUILD_ISO_CMD)
+            raise NotImplementedError
```

Comment:
hmm, so the way this is being used now is as a plugin, for example https://github.com/hashicorp/vagrant/blob/8e406184cb214a52996709283f7bd5dc29d0ac99/plugins/hosts/linux/plugin.rb#L14. So, you can call `host.capability(isofs_available)`. That is relying on extending this module, making this function available in the class as `self.isofs_available`. If a plugin capability does not set the `BUILD_ISO_CMD` it has used this module incorrectly. I think maybe this setup is not as straight forward as it could be. Moving this to the plugins capability will make the intention more clear